frontend/javascripts/oxalis/model/accessors/dataset_accessor.js

```diff
@@ -27,6 +27,123 @@ import { aggregateBoundingBox } from "libs/utils";
 import { formatExtentWithLength, formatNumberToLength } from "libs/format_utils";
 import messages from "messages";
 
+export type ResolutionsMap = Map<number, Vector3>;
+
+export class ResolutionInfo {
+  resolutions: Array<Vector3>;
+  resolutionMap: Map<number, Vector3>;
+
+  constructor(resolutions: Array<Vector3>) {
+    this.resolutions = resolutions;
+    this._buildResolutionMap();
+  }
+
+  _buildResolutionMap() {
+    // Each resolution entry can be characterized by it's greatest resolution dimension.
+    // E.g., the resolution array [[1, 1, 1], [2, 2, 1], [4, 4, 2]] defines that
+    // a zoomstep of 2 corresponds to the resolution [2, 2, 1] (and not [4, 4, 2]).
+    // Therefore, the largest dim for each resolution has to be unique across all resolutions.
+
+    // This function creates a map which maps from powerOfTwo (2**index) to resolution.
+
+    const resolutions = this.resolutions;
+
+    if (resolutions.length !== _.uniqBy(resolutions.map(_.max)).length) {
+      throw new Error("Max dimension in resolutions is not unique.");
+    }
+
+    this.resolutionMap = new Map();
+    for (const resolution of resolutions) {
+      this.resolutionMap.set(_.max(resolution), resolution);
+    }
+  }
+
+  getResolutionsWithIndices(): Array<[number, Vector3]> {
+    return Array.from(this.resolutionMap.entries()).map(entry => {
+      const [powerOfTwo, resolution] = entry;
+      const resolutionIndex = Math.log2(powerOfTwo);
+      return [resolutionIndex, resolution];
+    });
+  }
+
+  indexToPowerOf2(index: number): number {
+    return 2 ** index;
+  }
+
+  hasIndex(index: number): boolean {
+    const powerOfTwo = this.indexToPowerOf2(index);
+    return this.resolutionMap.has(powerOfTwo);
+  }
+
+  getResolutionByIndex(index: number): ?Vector3 {
+    const powerOfTwo = this.indexToPowerOf2(index);
+    return this.getResolutionByPowerOf2(powerOfTwo);
+  }
+
+  getResolutionByIndexOrThrow(index: number): Vector3 {
+    const resolution = this.getResolutionByIndex(index);
+    if (!resolution) {
+      throw new Error(`Resolution with in index ${index} does not exist`);
+    }
+    return resolution;
+  }
+
+  getResolutionByIndexWithFallback(index: number): Vector3 {
+    const resolutionMaybe = this.getResolutionByIndex(index);
+    if (resolutionMaybe) {
+      return resolutionMaybe;
+    } else {
+      const powerOf2 = this.indexToPowerOf2(index);
+      return [powerOf2, powerOf2, powerOf2];
+    }
+  }
+
+  getResolutionByPowerOf2(powerOfTwo: number): ?Vector3 {
+    return this.resolutionMap.get(powerOfTwo);
+  }
+
+  getHighestResolutionIndex(): number {
+    return Math.log2(this.getHighestResolutionPowerOf2());
+  }
+
+  getHighestResolutionPowerOf2(): number {
+    return _.max(Array.from(this.resolutionMap.keys()));
+  }
+
+  // todo: consider to only go upwards instead of both directions.
```

Comment:
For some cases, I'm not really sure whether it's the best thing to always go upwards. For example, it could be nice that the cell selection even works when the segmentation cannot be rendered (since this is only a shader limitation and the cell picking is done in JS). However, for the prefetching (as you say) it doesn't make sense to go in both directions.
I added another method (`getIndexOrClosestHigherIndex`) which only goes in the poorer mag direction and used it where appropriate. Let me know in case you disagree!